packages/messaging/lib/modular/index.d.ts

```diff
@@ -4,6 +4,10 @@ import { ReactNativeFirebase } from '@react-native-firebase/app';
 import Messaging = FirebaseStorageTypes.Module;
 import FirebaseApp = ReactNativeFirebase.FirebaseApp;
 
+export import AuthorizationStatus = FirebaseMessagingTypes.Statics.AuthorizationStatus;
```

Comment:
This one seems to be following the style of just exporting these statics in modular, and *not* exporting them in namespaced. Which means storage should probably have it's namespace export of it's statics removed

Then if that style works, all three of these "statics" type PRs can go in

Reply:
Hmmm, disagree on this one. We need to preserve the `FirebaseMessagingTypes.Statics.AuthorizationStatus` for users still using namespaced so they have correct types from static, and reexport as a top level import (i.e. modular). I don't think we should re-export as top level from namespaced type declaration file as it serves no purpose there.

When we come to delete the namespaced type declarations, we would lose it as a type for modular exports (although unit tests would fail at this point), it makes sense to me to put it where it belongs.

I have committed again because I found a better way to reexport and tested in unit tests the individual properties.